src/plugins/intel_cpu/tests/functional/single_layer_tests/matmul.cpp

```diff
@@ -169,6 +169,14 @@ class MatMulLayerCPUTest : public testing::WithParamInterface<MatMulLayerCPUTest
         function = makeNgraphFunction(netType, params, matMul, cpuNodeType);
         checkFusingPosition = false;
     }
+
+    bool primTypeCheck(std::string primType) const override {
+        // FullyConnected may use brgconv 1x1 kernel
+        if (cpuNodeType == "FullyConnected" && primType == "brgconv_avx512_1x1_FP32")
+            return true;
+
+        return selectedType == primType;
+    }
```

Comment:
Should not we set the expecting implementation type in advance, instead of using this implicit condition? In these tests we setting the expected parameters in accordance to the input, so that the correctness in tested properly. In such scenario, suppose we broke our one row shape skip condition, so that the plugin start to use brgemm for such shapes, we would like to detect such unexpected behavior, but with this implicit condition that would not be possible.

Reply:
Special cases `smoke_FC_2D_Brgconv1x1` for using brgconv1x1 added.